python/Ganga/GPIDev/Credentials/GridProxy.py

```diff
@@ -206,12 +204,10 @@ def chooseCommandSet(self):
         Return value: None
         """
         from Ganga.GPIDev.Lib.GangaList.GangaList import GangaList
-        if "ICommandSet" == self.command._name or isType(self.command, GangaList):
-            if self.voms:
-                self.command = self.vomsCommand
-            else:
-                self.command = self.gridCommand
-        return None
+        if self.voms:
```

Comment:
The problem at the moment is that it's cached the first time the object is created and then it can never be changed again. The problem here is that we have one interface (`GridProxy`) which can act in any number of ways depending on the config variables when Ganga was started. Admittedly the change here is only needed for running the tests but I feel that caching this is unnecessarily restrictive.  

Information about the path, voname and info() are all cached still.